app/uk/gov/hmrc/agentmapping/controller/MappingDetailsController.scala

```diff
@@ -73,10 +73,13 @@ class MappingDetailsController @Inject()(
   def transferSubscriptionRecordToMappingDetails(arn: Arn): Action[AnyContent] =
     authorisedWithProviderId { implicit request => implicit providerId =>
       subscriptionConnector.getUserMappings(AuthProviderId(providerId)).flatMap {
-        case Some(userMappings) =>
+        case Some(userMappings) if userMappings.nonEmpty =>
           val record = MappingDetailsRepositoryRecord(arn, userMappings2MappingDetails(userMappings))
           repository.create(record).map(_ => Created)
 
+        case Some(userMappings) if userMappings.isEmpty =>
+          Future successful Ok("No user mappings found")
```

Comment:
i think this case guard here is a bit redundant, but if you think it makes it more clear then that's fine